loopy/expression.py

```diff
@@ -76,7 +76,10 @@ def combine(vectorizabilities):
         return reduce(and_, vectorizabilities)
 
     def map_sum(self, expr):
-        return any(self.rec(child) for child in expr.children)
+        possible = False
+        for child in expr.children:
+            possible = self.rec(child) if True else possible
+        return possible
```

Comment:
I think these conditionals are equivalent, no?

Reply:
The problem with `any` is that if you recurse into the first child and it's vectorisable, it is assumed that the whole expression is vectorisable. Then it doesn't recurse into the second child which was containing a function call to `cos` in the failing test and the `VectorizabilityError` was not raised.